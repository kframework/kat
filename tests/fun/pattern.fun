datatype ('a,'b)    pair   = Pair   'a 'b
datatype ('a,'b,'c) triple = Triple 'a 'b 'c

letrec length = fun [       ] -> 0
                |   [ h : t ] -> 1 + length [ t ]
and   complex = fun (Triple [ Pair h1 h2 : t ] l  [ Pair a 2 : Pair 3 b : c ]) -> Pair (h2 + length [ t ] + b) a
                |   (Triple [                ] [] [ Pair 7 2 : x : c        ]) -> x
                |   default                                                    -> Pair 0 0
and  cons h [ t ] = [ h : t ]
and     map f = fun [       ] -> []
                |   [ h : t ] -> cons (f h) (map f [ t ])
in map complex [ Triple [ Pair 8 7 : .Exps ] [] [ Pair 9 2 : Pair 3 3 : Pair 2 2       : .Exps ]
               : Triple [                  ] [] [ Pair 7 2 : Pair 0 1 : Pair (-1) (-1) : .Exps ]
               : Triple [                  ] [] [                                              ]
               : .Exps
               ]

// [ Pair 10 9 : Pair 0 1 : Pair 0 0 : .Vals ]
