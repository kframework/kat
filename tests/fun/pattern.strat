5a1b24a8078d93f572402506b0b1104b orig
7f886956cb57d6ef2019fb435f972fd7 exec
