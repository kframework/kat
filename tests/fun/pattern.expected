
Strategy: orig
================================================================================
<FUN>
  <k>
    [ Pair 10 9 : Pair 0 1 : Pair 0 0 : .Vals ]
  </k>
  <env>
    .Map
  </env>
  <envs>
    .List
  </envs>
</FUN>

Strategy: exec
================================================================================
<kat-FUN>
  <s>
    #STUCK ( )
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <FUN>
      <k>
        [ Pair 10 9 : Pair 0 1 : Pair 0 0 : .Vals ]
      </k>
      <env>
        .Map
      </env>
      <envs>
        .List
      </envs>
    </FUN>
  </harness>
</kat-FUN>
